packages/techdocs-common/src/stages/publish/azureStorage.ts

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import path from 'path';
+import express from 'express';
+import {
+  BlobServiceClient,
+  BlobUploadCommonResponse,
+  StorageSharedKeyCredential,
+} from '@azure/storage-blob';
+import { Logger } from 'winston';
+import { Entity, EntityName } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { getHeadersForFileExtension, getFileTreeRecursively } from './helpers';
+import { PublisherBase, PublishRequest } from './types';
+
+export class AzureStoragePublish implements PublisherBase {
+  static async fromConfig(
+    config: Config,
+    logger: Logger,
+  ): Promise<PublisherBase> {
+    let account = '';
+    let accountKey = '';
+    let containerName = '';
+    try {
+      account = config.getString(
+        'techdocs.publisher.azureStorage.credentials.account',
+      );
+      accountKey = config.getString(
+        'techdocs.publisher.azureStorage.credentials.accountKey',
+      );
+      containerName = config.getString(
+        'techdocs.publisher.azureStorage.containerName',
+      );
+    } catch (error) {
+      throw new Error(
+        "Since techdocs.publisher.type is set to 'azureStorage' in your app config, " +
+          'credentials and containerName are required in techdocs.publisher.azureStorage ' +
+          'required to authenticate with Azure Storage.',
+      );
+    }
+
+    const credential = new StorageSharedKeyCredential(account, accountKey);
+    const storageClient = new BlobServiceClient(
+      `https://${account}.blob.core.windows.net`,
+      credential,
+    );
+
```

Comment:
Refer to my comment above about optional credentials in config. The PRs for AWS/GCS credentials will help!